python-exporters/microstrategy/exporter.py

```diff
@@ -115,7 +116,9 @@ def open(self, schema):
 
     def write_row(self, row):
         row_obj = {}
-        for (col, val) in zip(self.schema, row):
+        for (col, val, dtype) in zip(self.schema, row, self.dss_columns_types):
+            if (not val or (type(val)==float and np.isnan(val))) and dtype=='string':
+                val = ""
```

Comment:
I think the condition `not val` will be triggered when any numeric type has a zero value so zero values might be treated as empty - so this might cause a problem for datasets with zero values. Probably it would be better to specifically check for None or any other the other cases needed (aside from the floating point NaN case).

Reply:
This condition is only triggered if dtype is string. I have experimented with all combinations of `string`/`int` `storage types`/`meanings` leading to a string dtype. My conclusion is that, starting with a cell containing a 0, you can't have both a dtype=string and the python type of val being something else than str. So, starting with a cell containing 0, (not val) should never be true if dtype is string.

Reply:
That's great - I think that addresses my concern. It might still be good to put a comment about it to make it clear to anyone reading this.